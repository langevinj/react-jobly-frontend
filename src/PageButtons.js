import React from 'react'

//create the buttons that display on the bottoms of lists
function PageButtons({setPageNum = () => null, numPages = 1, pageNum = 0}){

    const prevPage = () => {
        setPageNum(pageNum => pageNum - 1)
    }

    const nextPage = () => {
        setPageNum(pageNum => pageNum + 1)
    }

    return(
        <div className="container buttons">
            {!pageNum > 0 ? <button className="btn-secondary disabled" onClick={prevPage} disabled>&#8592;</button> : <button className="btn-primary" onClick={prevPage}>&#8592;</button>}
            {pageNum === numPages - 1 ? <button onClick={nextPage} className="btn-secondary" disabled>&#8594;</button> : <button onClick={nextPage} className="btn-primary">&#8594;</button>}
        </div>
    )
}

export default PageButtons;