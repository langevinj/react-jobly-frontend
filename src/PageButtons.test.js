import React from 'react'
import { render } from '@testing-library/react'
import PageButtons from './PageButtons'
import { MemoryRouter } from 'react-router-dom'

it("renders without crashing", function() {
    render(<PageButtons />);
});

it("matches the snapshot", function() {
    const { asFragment } = render(<PageButtons />);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when on the last page", function() {
    const { asFragment } = render(<PageButtons numPages={5} pageNum={4}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot when on the first page", function() {
    const { asFragment } = render(<PageButtons numPages={5} pageNum={0}/>);
    expect(asFragment()).toMatchSnapshot();
});