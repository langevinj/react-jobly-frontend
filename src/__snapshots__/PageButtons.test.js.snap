// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when on the last page 1`] = `
<DocumentFragment>
  <div
    class="container buttons"
  >
    <button
      class="btn-primary"
    >
      ←
    </button>
    <button
      class="btn-secondary"
      disabled=""
    >
      →
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container buttons"
  >
    <button
      class="btn-secondary disabled"
      disabled=""
    >
      ←
    </button>
    <button
      class="btn-secondary"
      disabled=""
    >
      →
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches the snapshot when on the first page 1`] = `
<DocumentFragment>
  <div
    class="container buttons"
  >
    <button
      class="btn-secondary disabled"
      disabled=""
    >
      ←
    </button>
    <button
      class="btn-primary"
    >
      →
    </button>
  </div>
</DocumentFragment>
`;
